packages/okta-adapter/src/filters/policy_priority.ts

```diff
@@ -168,49 +168,86 @@ const getParentPolicy = (rule: InstanceElement): InstanceElement | undefined =>
   return getParents(rule)[0]?.value
 }
 
-const deployPriorityChange = async ({
+// use polling to overcome edge case where the policy is not yet updated
+const getWithPolling = async ({
+  args,
+  client,
+}: {
+  args: clientUtils.ClientBaseParams
+  client: clientUtils.HTTPWriteClientInterface & clientUtils.HTTPReadClientInterface
+}): Promise<clientUtils.Response<clientUtils.ResponseValue | clientUtils.ResponseValue[]>> => {
+  const pollingArgs: definitionsUtils.PollingArgs = {
+    interval: 1000,
+    retries: 3,
+    checkStatus: response => response.status === 200,
+    retryOnStatus: [404],
+  }
+  const clientGet = async ({
+    url,
+  }: clientUtils.ClientBaseParams): Promise<
+    clientUtils.Response<clientUtils.ResponseValue | clientUtils.ResponseValue[]>
+  > => client.get({ url })
+  const res = await clientUtils.executeWithPolling(args, pollingArgs, clientGet)
+  return res
+}
+
+const updatePriorityField = async ({
   client,
-  priority,
+  requiredPriority,
   instance,
-  apiDefinitions,
-  additionalUrlVars,
+  deployPolicyPath,
+  fieldsToOmit,
 }: {
   client: clientUtils.HTTPWriteClientInterface & clientUtils.HTTPReadClientInterface
-  priority: number
+  requiredPriority: number
   instance: InstanceElement
-  apiDefinitions: OktaSwaggerApiConfig
-  additionalUrlVars: Record<string, string>
+  deployPolicyPath: definitionsUtils.EndpointPath
+  fieldsToOmit: string[]
 }): Promise<void> => {
-  const { type } = instance.value
-  const ruleTypeName = instance.elemID.typeName
-  const baseData = { priority: setPriority(ruleTypeName, priority), type, name: instance.value.name }
-  // SignOn and MultiFactorEnrollment rules must include the `actions` field in the payload
-  const data = [SIGN_ON_RULE_TYPE_NAME, MFA_RULE_TYPE_NAME].includes(instance.elemID.typeName)
-    ? { ...baseData, actions: instance.value.actions }
-    : baseData
-  const typeDefinition = apiDefinitions.types[instance.elemID.typeName]
-  const deployRequest = typeDefinition.deployRequests ? typeDefinition.deployRequests.modify : undefined
-  const deployUrl = deployRequest?.url
-  if (deployUrl === undefined) {
-    throw new Error('Failed to deploy priority change due to missing url')
+  const pathContext = _.merge({}, instance.value, { parent_id: getParentPolicy(instance)?.value?.id })
```

Comment:
nit
```suggestion
const pathContext = { ...instance.value, parent_id: getParentPolicy(instance)?.value?.id }
```
I think that Using the spread operator { ...instance.value, parent_id: parentPolicyId } is better than _.merge({}, instance.value, { parent_id: ... }) because it simplifies the operation, avoids unnecessary deep merging, and guarantees immutability without needing an extra empty object.